File: Bai_11/intent-fliter-call-website/app/src/main/java/vn/edu/tlu/intent_fliter_call_website/MainActivity.java
package vn.edu.tlu.intent_fliter_call_website;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {
    Button btnSearch;
    EditText txtSearch;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });
        btnSearch = findViewById(R.id.btn_search);
        txtSearch = findViewById(R.id.edit_text_search);
        btnSearch.setOnClickListener(v -> {
            String search = txtSearch.getText().toString();
            if (!search.isEmpty()) {
                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(String.format("https:%s", search)));

                startActivity(intent);
            }else {
                Toast.makeText(this, "Please enter a search query", Toast.LENGTH_SHORT).show();
                return;
            }
        });
    }
}

File: Bai_2/app/src/main/java/com/example/bai_2/MainActivity.kt
package com.example.bai_2

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private lateinit var btnKetQua : Button
    private var soA: Int = 0
    private var soB: Int = 0
    private var ketQua: Float = 0.0f

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        // Lấy giá trị từ EditText và chuyển đổi sang kiểu Int
        btnKetQua = findViewById(R.id.btn_ket_qua)

        // Hiển thị kết quả sang txt_ket_qua trong ResultActivity
        btnKetQua.setOnClickListener {
            // try catch để bắt lỗi khi người dùng nhập vào không phải số
            soA = try {
                findViewById<EditText>(R.id.txt_a).text.toString().toInt()
            } catch (e: NumberFormatException) {
                Toast.makeText(this, "Số a vui lòng nhập số", Toast.LENGTH_SHORT).show()
                0
            }

            soB = try {
                findViewById<EditText>(R.id.txt_b).text.toString().toInt()
            } catch (e: NumberFormatException) {
                Toast.makeText(this, "Số b vui lòng nhập số", Toast.LENGTH_SHORT).show()
                0
            }

            // Giải phương trình bậc nhất
            ketQua = if (soA == 0) {
                if (soB == 0) {
                    Float.POSITIVE_INFINITY
                } else {
                    Float.NEGATIVE_INFINITY
                }
            } else {
                -soB.toFloat() / soA
            }
            val intent = Intent(this, ResultActivity::class.java)
            intent.putExtra("KET_QUA", ketQua)
            startActivity(intent)
        }
    }
}